src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/PackageVulnerabilityService.cs

```diff
@@ -0,0 +1,139 @@
+                if (result is null) continue;
+
+                if (result.KnownVulnerabilities != null)
+                {
+                    knownVulnerabilities ??= new();
+                    knownVulnerabilities.AddRange(result.KnownVulnerabilities);
+                }
+
+                if (result.Exceptions != null)
+                {
+                    if (errors == null)
+                    {
+                        errors = new();
+                    }
+
+                    errors.AddRange(result.Exceptions.InnerExceptions);
+                }
+            }
+
+            GetVulnerabilityInfoResult final =
+                knownVulnerabilities != null || errors != null
+                ? new(knownVulnerabilities, errors != null ? new AggregateException(errors) : null)
+                : null;
+            return final;
+        }
+
+        // Copied from NuGet.Commands.Restore.Utility.AuditUtility.GetKnownVulnerabilities
+        private static List<PackageVulnerabilityInfo> GetKnownVulnerabilities(
+            string name,
+            NuGetVersion version,
+            IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> knownVulnerabilities)
+        {
+            HashSet<PackageVulnerabilityInfo> vulnerabilities = null;
+
+            if (knownVulnerabilities == null) return null;
+
+            foreach (var file in knownVulnerabilities)
+            {
+                if (file.TryGetValue(name, out var packageVulnerabilities))
+                {
+                    foreach (var vulnInfo in packageVulnerabilities)
+                    {
+                        if (vulnInfo.Versions.Satisfies(version))
+                        {
+                            if (vulnerabilities == null)
+                            {
+                                vulnerabilities = new();
+                            }
+                            vulnerabilities.Add(vulnInfo);
+                        }
+                    }
+                }
+            }
+
+            return vulnerabilities != null ? vulnerabilities.ToList() : null;
+        }
+
+        private IEnumerable<PackageVulnerabilityMetadataContextInfo> ConvertToPackageVulnerabilityMetadataContextInfo(IEnumerable<PackageVulnerabilityInfo> packageVulnerabilities)
+        {
+            if (packageVulnerabilities == null || !packageVulnerabilities.Any())
+            {
+                return Enumerable.Empty<PackageVulnerabilityMetadataContextInfo>();
+            }
+
+            return packageVulnerabilities?.Select(pvi => new PackageVulnerabilityMetadataContextInfo(pvi.Url, (int)pvi.Severity));
+        }
+
+        private void ReplayErrors(AggregateException exceptions)
+        {
+            foreach (Exception exception in exceptions.InnerExceptions)
+            {
+                _logger.Log(ProjectManagement.MessageLevel.Warning, Strings.Error_VulnerabilityDataFetch, exception.Message);
```

Comment:
Where does this go? OutputWindow? 
Some of the loggers in this space log to the activity log which is not very helpful.

Reply:
@nkolev92 According to [this](https://github.com/NuGet/NuGet.Client/blob/545b3ce20365c91883518393c23988409988f20f/src/NuGet.Clients/NuGet.VisualStudio.Common/UserInterfaceService/INuGetUILogger.cs#L19-L23), it logs to the Error List for warnings and errors. This is logged as a warning. Do you think there's any action here or can this comment be resolved?